datumaro/plugins/cityscapes_format.py

```diff
@@ -0,0 +1,330 @@
+                segm_ids = np.unique(instances_mask)
+                for segm_id in segm_ids:
+                    if segm_id < 1000:
+                        semanticId = segm_id
+                        isCrowd = True
+                        id = segm_id
+                    else:
+                        semanticId = segm_id // 1000
+                        isCrowd = False
+                        id = segm_id % 1000
+                    anns.append(Mask(image=self._lazy_extract_mask(instances_mask, segm_id),
+                        label=semanticId, id=id,
+                        attributes = { 'is_crowd': isCrowd }))
+            items[sample_id] = DatasetItem(id=sample_id, subset=self._subset,
+                image=image_path, annotations=anns)
+        return items
+
+    @staticmethod
+    def _lazy_extract_mask(mask, c):
+        return lambda: mask == c
+
+    def _get_city_and_sample(self, full_path):
+        related_path = osp.relpath(full_path, self._path)
+        city_name = related_path.split('/')[0]
+
+        sample_id = osp.basename(full_path)
+        sample_id = sample_id.split(CityscapesPath.ORIGINAL_IMAGE)[0]
+
+        return city_name, sample_id
+
+
+class CityscapesImporter(Importer):
+    @classmethod
+    def find_sources(cls, path):
+        return cls._find_sources_recursive(path, '', 'cityscapes',
+            dirname=osp.join(CityscapesPath.IMGS_FINE_DIR,
+            CityscapesPath.ORIGINAL_IMAGE_DIR), max_depth=1)
+
+
+LabelmapType = Enum('LabelmapType', ['cityscapes', 'source'])
+
+class CityscapesConverter(Converter):
+    DEFAULT_IMAGE_EXT = '.png'
+
+    @staticmethod
+    def _get_labelmap(s):
+        if osp.isfile(s):
+            return s
+        try:
+            return LabelmapType[s].name
+        except KeyError:
+            import argparse
+            raise argparse.ArgumentTypeError()
+
+    @classmethod
+    def build_cmdline_parser(cls, **kwargs):
+        parser = super().build_cmdline_parser(**kwargs)
+
+        parser.add_argument('--apply-colormap', type=str_to_bool, default=True,
+            help="Use colormap for class masks (default: %(default)s)")
+        parser.add_argument('--label-map', type=cls._get_labelmap, default=None,
+            help="Labelmap file path or one of %s" % \
+                ', '.join(t.name for t in LabelmapType))
+        return parser
+
+    def __init__(self, extractor, save_dir,
+            apply_colormap=True, label_map=None, **kwargs):
+        super().__init__(extractor, save_dir, **kwargs)
+
+        self._apply_colormap = apply_colormap
+
+        if label_map is None:
+            label_map = LabelmapType.source.name
+        self._load_categories(label_map)
+
+    def apply(self):
+        os.makedirs(self._save_dir, exist_ok=True)
+
+        for subset_name, subset in self._extractor.subsets().items():
+            for item in subset:
+                item.id = item.id.replace('/', '_')
+                city_name = item.id.split('_')[0]
```

Comment:
I don't think such approach is correct because this way we change the layout of images the dataset. Probably, it is better to avoid this and just use what is given. I don't see how city name is used except for image organization, so I'd consider it just a relative path. Check `Converter._make_image_filename()` and `_save_image()` parameters.

Reply:
This part is rewritten, tests are updated